src/lib/eslint/TypescriptEslintStrategy.ts

```diff
@@ -168,75 +120,81 @@ export class TypescriptEslintStrategy implements EslintStrategy {
+		const tsconfigFromWorkingDirectory = await this.checkWorkingDirectoryForTsconfig();
+		const tsConfigFromOptions = await this.checkEngineOptionsForTsconfig(engineOptions);
+
+		if (!tsconfigFromWorkingDirectory && !tsConfigFromOptions) {
+			// Unable to find in current directory and not specified in the engineOptions
+			throw SfdxError.create('@salesforce/sfdx-scanner', 'TypescriptEslintStrategy', 'MissingTsConfigFromCwd',
+				[TS_CONFIG, cwd, TS_CONFIG]);
+		} else if (tsconfigFromWorkingDirectory && tsConfigFromOptions) {
+			// Found in current directory and specified in the engineOptions
+			throw SfdxError.create('@salesforce/sfdx-scanner', 'TypescriptEslintStrategy', 'MultipleTsConfigs',
+				[TS_CONFIG, tsconfigFromWorkingDirectory, tsConfigFromOptions]);
 		}
 
-		this.logger.trace(`Using ${TS_CONFIG} from ${tsconfigPath}`);
-		return tsconfigPath;
-	}
+		const foundTsConfig = tsconfigFromWorkingDirectory || tsConfigFromOptions;
 
-	private async getTsconfigFromConfig(): Promise<string> {
-		let tsconfigPath = NO_TS_CONFIG;
-		// Check if overriddenConfig is available
-		const overriddenConfig = this.config.getOverriddenConfigPath(ENGINE.ESLINT_TYPESCRIPT);
-		if (!overriddenConfig) {
-			this.logger.trace(`Did not find an overridden path from Config`);
-			return tsconfigPath;
-		}
+		this.logger.trace(`Using ${TS_CONFIG} from ${foundTsConfig}`);
 
-		// Complain if overriddenConfig does not exist.
-		if (!(await this.fileHandler.exists(path.resolve(overriddenConfig)))) {
-			throw SfdxError.create('@salesforce/sfdx-scanner', 'eslintEngine', 'InvalidPath', [overriddenConfig]);
-		}
+		return foundTsConfig;
+	}
 
-		// If overriddenConfig is a directory, look for tsconfig.json file.
-		if (await this.fileHandler.isDir(overriddenConfig)) {
-			this.logger.trace(`Looking for ${TS_CONFIG} inside overridden dir from config ${overriddenConfig}`);
-			tsconfigPath = await this.checkDirectoryForTsconfig(overriddenConfig);
-			// Complain if we can't find a tsconfig.json inside.
-			if (!tsconfigPath) {
-				throw new SfdxError(`Overridden path [${overriddenConfig}] does not contain ${TS_CONFIG}`);
+	/**
+	 * Returns the tsconfig.json file specified in the engineOptions if it is specified and valid, or null if
+	 * engineOptions doesn't specify a tsconfig.json value.
+	 * Throw an error if the file specified in engineOptions is invalid for any reason.
+	 */
+	protected async checkEngineOptionsForTsconfig(engineOptions: Map<string, string>): Promise<string> {
+		const tsConfigFromOptions = engineOptions.get(TYPESCRIPT_ENGINE_OPTIONS.TSCONFIG);
+
+		if (tsConfigFromOptions != null) {
+			if (!(await this.fileHandler.exists(tsConfigFromOptions))) {
+				// Specified in the engineOptions but it isn't a file
+				throw SfdxError.create('@salesforce/sfdx-scanner', 'TypescriptEslintStrategy', 'NotAFileTsConfigFromOptions',
+					[TS_CONFIG, tsConfigFromOptions]);
+			} else if (path.basename(tsConfigFromOptions).toLowerCase() !== TS_CONFIG) {
```

Comment:
Do we know if typescript/eslint recognizes a tsconfig.json file named in mixed case? Such as "TsConfig.json" or "tsConfig.json"? I remember reading somewhere that eslint is strict about using only a file named exactly as "tsconfig.json", but not sure about how it handles a mixed-case name.

Reply:
Oh..I didn't know that was a hard requirement. I put in the toLowerCase to avoid false errors for file systems that are case insensitive. Should the toLowerCase be removed?

Reply:
Actually, good point, renaming the file to TsConfig.json causes another part of code to fail. I get `Unable to find 'tsconfig.json' in current directory` if I don't specify --tsconfig. However specifying --tsconfig TsConfig.json works, so eslint doesn't seem to have an issue with it.

Reply:
Sounds good then 👍